### PR TITLE
Upgrade erlang.mk, fix dialyzer warnings and other goodies

### PR DESCRIPTION
**Pull request by elbrujohalcon**
Basically, getting ready for inaka/erlang-katana#89 :muscle:

### CODE REVIEW

File: src/ktn_user_default.erl [hunk 1]
```diff
@@ -2,11 +2,16 @@
 -export([
          xref/0
          ,l_all/1
```
Comment (gadgetci) [inline src/ktn_user_default.erl]: According to **Elvis**:
> Missing space after "," on line 4

Comment (gadgetci) [inline src/ktn_user_default.erl]: According to **Elvis**:
> Missing space after "," on line 4

File: src/ktn_recipe.erl [hunk 2]
```diff
@@ -182,6 +178,7 @@ pretty_print_normalized(NormalizedTransitions) ->
 %% correct arity.
 %% It does not verify structural properties of the FSM defined by the transition
 %% table (e.g. connected, acyclic, arboreal).
+-spec verify(atom() | transitions()) -> term().
```
Comment (igaray) [inline src/ktn_recipe.erl]: This spec should be module() | ... too :)

Comment (elbrujohalcon) [inline src/ktn_recipe.erl]: I'm fixing it in the next PR

**Pull request merged**